### PR TITLE
Hotfix/lumberjack

### LINKED ISSUE #1227: Sawmill lvl 2 named Baker 2
Minecolonies-universal-1.10.2-0.8.2817

 Slight naming error.

Edit:  The lvl 3 Sawmill is named Baker 3.

### LINKED ISSUE #1229: Lumberjack GUI not opening
### Minecolonies version

version 1.10.2-0.8.2817

### Expected behavior

Buildable Hut and openable GUI for Lumberjack

### Actual behaviour

[12:12:57] [Server thread/ERROR] [minecolonies]: Unknown Building type 'com.minecolonies.coremod.tileentities.TileEntityColonyBuilding' or missing constructor of proper format.
java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_121]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_121]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source) ~[?:1.8.0_121]
        at java.lang.reflect.Constructor.newInstance(Unknown Source) ~[?:1.8.0_121]
        at com.minecolonies.coremod.colony.buildings.AbstractBuilding.create(AbstractBuilding.java:382) ~[AbstractBuilding.class:?]
        at com.minecolonies.coremod.colony.Colony.addNewBuilding(Colony.java:1605) ~[Colony.class:?]
        at com.minecolonies.coremod.blocks.AbstractBlockHut.func_180633_a(AbstractBlockHut.java:262) ~[AbstractBlockHut.class:?]
        at com.minecolonies.coremod.blocks.AbstractBlockHut.onBlockPlacedByBuildTool(AbstractBlockHut.java:291) ~[AbstractBlockHut.class:?]
        at com.minecolonies.coremod.network.messages.BuildToolPlaceMessage.handleHut(BuildToolPlaceMessage.java:174) ~[BuildToolPlaceMessage.class:?]
        at com.minecolonies.coremod.network.messages.BuildToolPlaceMessage.messageOnServerThread(BuildToolPlaceMessage.java:133) ~[BuildToolPlaceMessage.class:?]
        at com.minecolonies.coremod.network.messages.BuildToolPlaceMessage.messageOnServerThread(BuildToolPlaceMessage.java:36) ~[BuildToolPlaceMessage.class:?]
        at com.minecolonies.coremod.network.messages.AbstractMessage.lambda$onMessage$0(AbstractMessage.java:24) ~[AbstractMessage.class:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_121]
        at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_121]
        at net.minecraft.util.Util.func_181617_a(SourceFile:45) [h.class:?]
        at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:668) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:387) [ld.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:613) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471) [MinecraftServer.class:?]
        at java.lang.Thread.run(Unknown Source) [?:1.8.0_121]
Caused by: java.lang.NoSuchMethodError: net.minecraft.item.Item.func_150895_a(Lnet/minecraft/item/Item;Lnet/minecraft/creativetab/CreativeTabs;Ljava/util/List;)V
        at com.minecolonies.coremod.colony.buildings.BuildingLumberjack.calcSaplings(BuildingLumberjack.java:129) ~[BuildingLumberjack.class:?]
        at com.minecolonies.coremod.colony.buildings.BuildingLumberjack.<init>(BuildingLumberjack.java:91) ~[BuildingLumberjack.class:?]
        ... 20 more
[12:12:57] [Server thread/ERROR] [minecolonies]: Colony 1 unable to create AbstractBuilding for class com.minecolonies.coremod.blocks.BlockHutLumberjack at BlockPos{x=365, y=67, z=238}
[12:12:57] [Server thread/ERROR] [minecolonies]: BuildTool: building is null!

### Steps to reproduce the problem

1. set block (with wand)
2. try to open
3. fail
4. try not to cry
5. cry alot

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #1227
Closes #1229

Fix lumberjack bug with side-only
fix warehouse and lumberjack guis

Review please

### CODE REVIEW

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 28 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 27 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L267](https://github.com/Minecolonies/minecolonies/blob/ba445ac354fec62ec9646036883bbd885b5798df/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L267): Reduce this class from 70 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/ba445ac354fec62ec9646036883bbd885b5798df/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/ba445ac354fec62ec9646036883bbd885b5798df/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/ba445ac354fec62ec9646036883bbd885b5798df/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/ba445ac354fec62ec9646036883bbd885b5798df/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/ba445ac354fec62ec9646036883bbd885b5798df/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [BuildingLumberjack.java#L36](https://github.com/Minecolonies/minecolonies/blob/ba445ac354fec62ec9646036883bbd885b5798df/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L36): Override the "equals" method in this class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/ba445ac354fec62ec9646036883bbd885b5798df/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/ba445ac354fec62ec9646036883bbd885b5798df/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/ba445ac354fec62ec9646036883bbd885b5798df/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 28 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 27 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L267](https://github.com/Minecolonies/minecolonies/blob/092e18951f5a9acb236de4b50c603c67e91b70db/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L267): Reduce this class from 70 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/092e18951f5a9acb236de4b50c603c67e91b70db/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/092e18951f5a9acb236de4b50c603c67e91b70db/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/092e18951f5a9acb236de4b50c603c67e91b70db/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/092e18951f5a9acb236de4b50c603c67e91b70db/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/092e18951f5a9acb236de4b50c603c67e91b70db/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [BuildingLumberjack.java#L36](https://github.com/Minecolonies/minecolonies/blob/092e18951f5a9acb236de4b50c603c67e91b70db/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L36): Override the "equals" method in this class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/092e18951f5a9acb236de4b50c603c67e91b70db/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/092e18951f5a9acb236de4b50c603c67e91b70db/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/092e18951f5a9acb236de4b50c603c67e91b70db/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 import com.minecolonies.api.util.constant.ToolType;
 import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.blockout.Log;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java]: MINOR Remove this unused import 'com.minecolonies.blockout.Log'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 29 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 28 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L249](https://github.com/Minecolonies/minecolonies/blob/74e0ebd931f1d7d8269c854ab18d2772318dc6bf/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L249): Reduce this class from 70 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/74e0ebd931f1d7d8269c854ab18d2772318dc6bf/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/74e0ebd931f1d7d8269c854ab18d2772318dc6bf/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/74e0ebd931f1d7d8269c854ab18d2772318dc6bf/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/74e0ebd931f1d7d8269c854ab18d2772318dc6bf/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/74e0ebd931f1d7d8269c854ab18d2772318dc6bf/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [BuildingLumberjack.java#L36](https://github.com/Minecolonies/minecolonies/blob/74e0ebd931f1d7d8269c854ab18d2772318dc6bf/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L36): Override the "equals" method in this class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/74e0ebd931f1d7d8269c854ab18d2772318dc6bf/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/74e0ebd931f1d7d8269c854ab18d2772318dc6bf/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/74e0ebd931f1d7d8269c854ab18d2772318dc6bf/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by xavierh — APPROVED** [review]

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 28 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 27 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BuildingLumberjack.java#L248](https://github.com/Minecolonies/minecolonies/blob/3ea5e7c5a5da5aaf271bb102a79b08ab921bca76/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L248): Reduce this class from 70 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/3ea5e7c5a5da5aaf271bb102a79b08ab921bca76/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/3ea5e7c5a5da5aaf271bb102a79b08ab921bca76/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/3ea5e7c5a5da5aaf271bb102a79b08ab921bca76/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/3ea5e7c5a5da5aaf271bb102a79b08ab921bca76/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/3ea5e7c5a5da5aaf271bb102a79b08ab921bca76/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [BuildingLumberjack.java#L35](https://github.com/Minecolonies/minecolonies/blob/3ea5e7c5a5da5aaf271bb102a79b08ab921bca76/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java#L35): Override the "equals" method in this class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/3ea5e7c5a5da5aaf271bb102a79b08ab921bca76/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/3ea5e7c5a5da5aaf271bb102a79b08ab921bca76/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/3ea5e7c5a5da5aaf271bb102a79b08ab921bca76/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**